apdet-hanna/src/apdet.cpp

```diff
@@ -469,144 +463,118 @@ int main()
+            break;
+        }
 
-        // /* Get pressure and temperature */
-        // retval = barometerGetAndLog(&curr_pres, &curr_temp);
-        // if (retval != STATUS_OK) {
-        //     break;
-        // }
+        /* Get pressure and temperature */
+        retval = barometerGetAndLog(&curr_pres, &curr_temp);
+        if (retval != STATUS_OK) {
+            break;
+        }
 
-        // /* Get current relative altitude (height above ground) */
-        // calcHeight(curr_pres, base_alt, &curr_height);
+        /* Get current relative altitude (height above ground) */
+        calcHeight(curr_pres, baseVars.base_alt, &curr_height);
 
 
         switch(curr_state) {
+
             case APDET_STATE_TESTING:
             {
-                /* Get acceleration */
-                status_t retval = accelerometerGetAndLog(&accel_x, &accel_y, &accel_z);
-                if (retval != STATUS_OK) {
-                    break;
-                }
-                accelGetMagnitudeAndLog(accel_x, accel_y, accel_z, &accel);
-
-                /* Get pressure and temperature */
-                retval = barometerGetAndLog(&curr_pres, &curr_temp);
-                if (retval != STATUS_OK) {
-                    break;
-                }
-
-                baseVars.base_pres = curr_pres;
-                baseVars.base_temp = curr_temp;
-
-                if (testStandby(accel, baseVars.base_alt)) {
-
-                    /* Update base values */
+                if (testStandby(accel, curr_height)) {
+                    /* Update and write pressure and temperature */
+                    baseVars.base_pres = curr_pres;
+                    baseVars.base_temp = curr_temp;
                     writeBaseVars(baseVars);
+                    /* Change state */
                     changeStateAndResetChecks(APDET_STATE_STANDBY, &curr_state,
                       state_change_check_arr, ARR_SIZE, &state_change_check_idx);
                 } else {
+                    /* recover base variables and state */
                     recoverAll(&curr_state, &baseVars);
                 }
                 break;
             }
 
             case APDET_STATE_STANDBY:
             {
-                /* Get acceleration */
-                status_t retval = accelerometerGetAndLog(&accel_x, &accel_y, &accel_z);
-                if (retval != STATUS_OK) {
-                    break;
-                }
-
-                /* Calculate magnitude of acceleration */
-                accelGetMagnitudeAndLog(accel_x, accel_y, accel_z, &accel);
-
                 if (detectLaunch(accel)) {
+
                     /* 1 means we have a passing check */
                     state_change_check_arr[state_change_check_idx] = 1;
-                    /* increment index with wrap-around */
-                    state_change_check_idx = (state_change_check_idx + 1) % ARR_SIZE;
+
                     if (sumArrElems(state_change_check_arr, ARR_SIZE) >= NUM_CHECKS) {
```

Comment:
The extra FOR-loop in this function could be optimized out by keeping track of the sum, and adding/ subtracting 1's and 0's to/from it - but this approach is good for readability and ok because the array is small